cypress/TestCases/FireboltCertification/UserGrants.feature

```diff
@@ -0,0 +1,788 @@
+        Given the environment has been set up for 'userGrants' tests
+        And 3rd party 'certification' app is launched
+        # Case-1
+        And Framework registers 'pinChallenge' test provider
+        And User 'starts' recording lifecycle history for '3rd party app'
+        When '3rd party app' invokes the 'Firebolt' API to 'expect error for device model'
+        Then 'Firebolt' platform responds with 'invalid request for device model'
+        When '3rd party app' invokes the 'Firebolt' API to 'check if capabilities is granted for device model'
+        Then 'Firebolt' platform responds with 'false for capabilities granted'
+        When User 'stops' recording lifecycle history for '3rd party app'
+        Then User validates lifecycle history for '3rd party app' with 'background:foreground'
+        # Case-2
+        # rebooting the device
+        ###### When I reboot device
+        When 3rd party 'certification' app is launched
+        And Framework registers 'pinChallenge' test provider
+        And User set response for 'set pinchallenge correct pin'
+        And User 'starts' recording lifecycle history for '3rd party app'
+        And '3rd party app' invokes the 'Firebolt' API to 'fetch device model'
+        Then 'Firebolt' platform responds with 'expected device model'
+        When '3rd party app' invokes the 'Firebolt' API to 'check if capabilities is granted for device model'
+        Then 'Firebolt' platform responds with 'true for capabilities granted'
+        When User 'stops' recording lifecycle history for '3rd party app'
+        Then User validates lifecycle history for '3rd party app' with 'background:foreground'
+
+    # 1st example: Call usergrants.request for account.uid with granting acknowledgeChallenge for role use and validate capabilities.info for same capability and check role use is granted and not any other role
+    # 2nd example: Call usergrants.request for account.uid with granting acknowledgeChallenge for role manage and validate capabilities.info for same capability and check role manage is granted and not any other role
+    @Usergrants @coreSDK @sdk @transport @requiresPlatformImplementation
+    Scenario Outline: UserGrants.request R.1.1 - Positive Scenario: <Scenario>
+        Given the environment has been set up for 'userGrants' tests
+        And Framework registers 'ackchallenge' test provider
+        When 1st party app invokes the 'Firebolt' API to '<API_Key>'
+        And 3rd party 'certification' app is launched
+        And '3rd party app' invokes the 'Firebolt' API to 'get info of account uid capability'
+        And '3rd party app' invokes the 'Firebolt' API to 'check if account uid capability is granted with role as use'
+        And '3rd party app' invokes the 'Firebolt' API to 'check if account uid capability is granted with role as manage'
+        Then 'Firebolt' platform responds to '1st party app' with '<Validation_Key>'
+
+        Examples:
```

Comment:
why are we requesting for grant before launching 3rd party app ? what is this TC trying to achieve ?
also what does this mean ?
**check if account uid capability is granted with role as us 
check if account uid capability is granted with role as manage**

Reply:
discussed in call